File: WorkoutTrackerAPI/Data/Models/WorkoutModels/Workout.cs
using Microsoft.EntityFrameworkCore;
using WorkoutTrackerAPI.Data.Models.WorkoutModels;

namespace WorkoutTrackerAPI.Data.Models;

public class Workout : WorkoutModel
{
    public string? Description { get; set; }
    public DateTime Created { get; set; }
    public bool IsPinned { get; set; }
    public int CountOfTrainings { get; set; }

    public string UserId { get; set; } = null!;
    public User? User { get; set; }

    public IEnumerable<ExerciseSetGroup>? ExerciseSetGroups { get; set; }
    public IEnumerable<WorkoutRecord>? WorkoutRecords { get; set; }
}

File: WorkoutTracker/src/app/workouts/workout.ts
import { WorkoutModel } from "src/app/shared/models/workout-model";
import { ModelWeight } from "../shared/models/model-weight";
import { Exercise } from "../exercises/models/exercise";
import { ExerciseSetGroup } from "../shared/models/exercise-set-group";

export interface Workout extends WorkoutModel {
    description: string|null;
    created: Date;
    started: Date|null;
    isPinned: boolean;
    weight: ModelWeight;

    exercises: Exercise[];
    exerciseSetGroups: ExerciseSetGroup[];
}